pkg/cache/application.go

```diff
@@ -599,6 +618,19 @@ func (app *Application) handleReleaseAppAllocationEvent(event *fsm.Event) {
 			}
 		}
 	}
+	app.postAppReleased()
+}
+
+func (app *Application) postAppReleased() {
+	canRunApplication := true
+	for _, task := range app.taskMap {
+		if !utils.IsPodTerminated(task.GetTaskPod()) {
+			canRunApplication = false
+		}
```

Comment:
when the shim deletes the placeholder pod, the pod will be gone, I don’t think this is a safe way to check.
instead of doing this check in the shim side, can we leverage the core side app state?
IIUC, the core side has the check when all the placeholders are removed, then moves the app to the Running state. In that case, can we simply rely on that, if the app state from the core transits to the Running state, and the shim is not in Running state yet, we trigger a RunApplication event.

Reply:
> when the shim deletes the placeholder pod, the pod will be gone, I don’t think this is a safe way to check.
> instead of doing this check in the shim side, can we leverage the core side app state?
> IIUC, the core side has the check when all the placeholders are removed, then moves the app to the Running state. In that case, can we simply rely on that, if the app state from the core transits to the Running state, and the shim is not in Running state yet, we trigger a RunApplication event.

IIU @kingamarton's earlier comments and offline discussions, shim is the right place to ensure all placeholders has been cleaned up correctly and app state would be "reserving" at this point in shim side. So, tried to make changes in that direction as depending blindly on core state doesn't ensure all placeholders has been cleaned up completely. core simply make the state as "resuming" as soon as it sees scheduling style as "soft" when placeholders being timed out. @kingamarton Can you also share your thoughts?